### DISCUSSION TITLE
How to fix - "no valid jpeg for X (set primary)"?

### DISCUSSION THREAD

**Discussion by tootbrute** [photoprism/photoprism#2038] [Q&A] [answered]
What do these messages in the log  mean? how can I "fix" them?

time="2022-02-08T16:15:33Z" level=info msg="index: found no valid jpeg for pqm0w611fihcelz2 (set primary)"
time="2022-02-08T16:15:33Z" level=info msg="index: found no valid jpeg for pqpfwnj2hjcwq5gc (set primary)"
time="2022-02-08T16:15:33Z" level=info msg="index: found no valid jpeg for pqpfwnmqctx6r3ke (set primary)"
time="2022-02-08T16:15:33Z" level=info msg="index: found no valid jpeg for pqpfwnp2ri90cfdj (set primary)"
time="2022-02-08T16:15:33Z" level=info msg="index: found no valid jpeg for pqpfwns31zb876xo (set primary)"
time="2022-02-08T16:15:33Z" level=info msg="index: found no valid jpeg for pqpfwnuysjtqsi4g (set primary)"

**Comment by lastzero** [comment, marked as answer]
There are many reasons why JPEGs can be missing:

https://docs.photoprism.app/getting-started/troubleshooting/#missing-pictures

**Reply by tootbrute** [reply]
Ok that gives me a lot of stuff to investigate, but how do I actually hunt down the specific files.

time="2022-04-08T15:50:38Z" level=info msg="index: found no valid jpeg for pr1h0821a5nsd490 (set primary)"

For example for this specific file, I search "pr1h0821a5nsd490" in Photoprism and it gives no result.

How do I hunt down this file so I can figure out how to fix it?

Can the logging give us some more specific information that can help us locate the files in question?

Or is there another way to search for this "pr1h0821a5nsd490" file?

**Reply by lastzero** [reply]
Did you try the uid: filter yet? Maybe in the Missing section as the picture likely has no valid JPEG.

**Reply by tootbrute** [reply]
Ah wonderful. For other dense people like me...

uid:pr1h0821a5nsd490 found the file.

Now I need to analyze it and see how to fix it. It's an AVC1, MP4.

Your guess seems correct though. There is no JPG thumbnail.
